Sources/WalletConnectAuth/AuthClient.swift

```diff
@@ -330,5 +330,8 @@ public final class AuthClient {
         pairingEngine.onProposeResponse = { [unowned self] sessionTopic in
             sessionEngine.setSubscription(topic: sessionTopic)
         }
+        pairingEngine.onApprovalResponse = { [unowned self] proposal in
+            sessionEngine.settlingProposal = proposal
+        }
```

Comment:
do you need that extra callback `onApprovalResponse` ? `onProposeResponse` is executed in the same function.

would it work to have two arguments(if needed) in on callback?

Reply:
Yes, changed the callback to 2 args